crates/tinymist/src/tool/preview/auth.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::{Context, Result};
+use futures::SinkExt;
+use futures::StreamExt;
+use hyper_tungstenite::{tungstenite::Message, HyperWebsocketStream};
+use rand::distributions::{Alphanumeric, DistString};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha512};
+
+const TOKEN_LENGTH: usize = 50;
+
+pub fn generate_token() -> String {
+    Alphanumeric.sample_string(&mut rand::thread_rng(), TOKEN_LENGTH)
```

Comment:
I noticed that in auth.ts, we generate crypto random numbers. Should we do the same thing here?

Reply:
I believe `rand::thread_rng` does provide cryptographically secure random numbers, see e.g. https://rust-random.github.io/book/guide-rngs.html